hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/RecoveredReplicationSource.java

```diff
@@ -45,18 +46,15 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(RecoveredReplicationSource.class);
 
-  private Path walDir;
-
   private String actualPeerId;
 
   @Override
-  public void init(Configuration conf, FileSystem fs, Path walDir, ReplicationSourceManager manager,
-    ReplicationQueueStorage queueStorage, ReplicationPeer replicationPeer, Server server,
-    String peerClusterZnode, UUID clusterId, WALFileLengthProvider walFileLengthProvider,
-    MetricsSource metrics) throws IOException {
-    super.init(conf, fs, walDir, manager, queueStorage, replicationPeer, server, peerClusterZnode,
-      clusterId, walFileLengthProvider, metrics);
-    this.walDir = walDir;
+  public void init(Configuration conf, FileSystem fs, Path walDir,
+    ReplicationSourceOverallController overallController, ReplicationQueueStorage queueStorage,
```

Comment:
There is only one ReplicationSourceManager instance in a RS? If so, can it carry the Controller rather than as here where it is another param on this init method?

Reply:
Yes. Only one ReplicationSourceManager and only one Controller, too. Only pass Controller to ReplicationSource. Then decouple ReplicationSourceManager and ReplicationSource.